nautobot_golden_config/models.py

```diff
@@ -177,32 +178,39 @@ def _verify_get_custom_compliance_data(compliance_details):
 
 def _get_hierconfig_remediation(obj):
     """Returns the remediating config."""
-    hierconfig_os = obj.device.platform.network_driver_mappings["hier_config"]
+    hierconfig_os = obj.device.platform.network_driver_mappings.get("hier_config")
+
     if not hierconfig_os:
-        raise ValidationError(f"platform {obj.network_driver} is not supported by hierconfig.")
+        raise ValidationError(f"platform {obj.device.platform.network_driver} is not supported by hierconfig.")
```

Comment:
While testing this function, I noticed that `obj.network_driver` is not the correct syntax, it should be `obj.device.platform.network_driver`. This would cause a separate exception that is not handled.